uint32 sonar0
uint32 sonar1
uint32 sonar2
uint32 sonar3
uint32 sonar4
uint32 sonar5
